src/App.tsx

```diff
@@ -1,16 +1,22 @@
 import './App.scss';
+import { useState } from 'react';
 import { MoviesList } from './components/MoviesList';
 import { NewMovie } from './components/NewMovie';
 import moviesFromServer from './api/movies.json';
+import { Movie } from './types/Movie';
 
 export const App = () => {
+  const [movies, setMovies] = useState(moviesFromServer);
+
+  const addMovie = (movie: Movie) => setMovies([...movies, movie]);
```

Comment:
If you want to update the state, you should use prevState 
https://reactjs.org/docs/hooks-reference.html#functional-updates
```suggestion
  const addMovie = (movie: Movie) => setMovies(prevState => [
    ...prevState, 
    movie,
  ]);
```